### PR TITLE
CLAP running on Summit

### PR DESCRIPTION
**Pull request by lukewys**
- Add multi-node support
 - Add multi-GPU / multi-node evaluation support
 - Add some codes and comments for debug
 - Clean up code and parameters a bit
 - Add additional parameters for no evaluation, only training

### CODE REVIEW

**Comment by rom1504** [comment]
Nice!

I highly recommend you use gradient checkpointing https://github.com/mlfoundations/open_clip/pull/85/files
Allows you to do x10 on batch size

**Comment by lukewys** [comment]
> Nice!
> 
> I highly recommend you use gradient checkpointing https://github.com/mlfoundations/open_clip/pull/85/files Allows you to do x10 on batch size

Wow! Sounds great! 
If I understand correctly, the gradient checkpointing you are pointing here is using a smarter way to backprop, and it still has one forward run per backward right? In contrast to some methods for saving GPU memory that basically forward multiple times with small batch size and backward once. (I cannot remember what the term for the latter, I was confused it with gradient checkpointing.)

Also, does that compatible with fp16 training?

**Comment by rom1504** [comment]
Yeah, what you are referring to is called gradient accumulation
Gradient checkpointing recomputes activation during backward which allows to spare some memory during forward
